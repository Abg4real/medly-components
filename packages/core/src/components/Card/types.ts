import { HTMLProps, WithThemeProp } from '@medly-components/utils';

export type ItemsPosition = 'top' | 'bottom' | 'left' | 'right' | 'center';
export type Variants = 'outlined' | 'solid' | 'flat';
export type FlowDirection = 'vertical' | 'horizontal';

export interface Props extends HTMLProps<HTMLDivElement>, WithThemeProp {
    /** To be used if rendering like Link */
    to?: string;
    /** Card design */
    variant?: 'outlined' | 'solid' | 'flat';
    /** Direction of the children */
    flowDirection?: 'vertical' | 'horizontal';
    /** Separotor between each children */
    separator?: boolean;
    /** To be used to difine width/height of cards inside cards */
    flex?: number;
    /** Enable padding */
    withPadding?: boolean;
    /** Change the cursor pointer */
    clickable?: boolean;
    /** Alignment of the card content */
    alignItems?: 'top' | 'bottom' | 'left' | 'right' | 'center';
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Takes full height of the parent component */
    fullHeight?: boolean;
}
