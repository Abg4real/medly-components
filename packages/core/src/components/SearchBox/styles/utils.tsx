import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
