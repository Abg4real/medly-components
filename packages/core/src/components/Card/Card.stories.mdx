import { HTMLProps } from '@medly-components/utils';
import { Card } from './Card.tsx';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Card.stories';

<Meta title="Core/Card" component={Card} />

# Card

### Simple card component with lot of customization.

You can play with the props in canvas tab.

<Preview withToolbar>
    <Story name="Basic">{stories.Basic()}</Story>
</Preview>

### Variants

<Preview>
    <Card variant="solid">
        <Text>Solid Card</Text>
    </Card>
    <Card variant="outlined">
        <Text>Outlined Card</Text>
    </Card>
    <Card variant="flat">
        <Text>Flat Card</Text>
    </Card>
</Preview>

### Children direction

<Preview>
    <Card variant="outlined" flowDirection="horizontal">
        <Text>1</Text>
        <Text>Horizontal</Text>
    </Card>
    <Card variant="outlined" flowDirection="vertical">
        <Text>1</Text>
        <Text>Vertical</Text>
    </Card>
</Preview>

### Props

<Props of={Card} />

### CardTheme

<Props of={stories.ThemeInterface} />
