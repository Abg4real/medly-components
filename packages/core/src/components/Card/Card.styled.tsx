import { defaultTheme } from '@medly-components/theme';
import { css, styled } from '@medly-components/utils';
import { Props } from './types';

export const CardStyled = styled('div').attrs(({ theme: { card } }) => ({ ...card }))<Props>`
    display: inline-flex;
    width: ${({ width }) => (width ? `${width}` : '250px')};
    height: ${({ height }) => (height ? `${height}` : '250px')};
    background-color: ${({ bgColor }) => bgColor};
    border-radius: 4px;
    box-shadow: 0 2px 4px 0 ${({ shadowColor }) => shadowColor};
    padding: 10px;
    border: 1px solid ${({ borderColor }) => borderColor};

    ${({ center }) =>
        center &&
        css`
            flex-direction: column;
            justify-content: center;
            align-items: center;
        `}
`;

CardStyled.defaultProps = {
    theme: defaultTheme
};
