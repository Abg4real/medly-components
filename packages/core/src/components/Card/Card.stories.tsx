import { CardTheme, defaultTheme } from '@medly-components/theme';
import { boolean, select } from '@storybook/addon-knobs';
import React from 'react';
import Text from '../Text';
import { Card } from './Card';
import { FlowDirection, ItemsPosition, Variants } from './types';

const variants: Variants[] = ['outlined', 'solid', 'flat'];
const itemsPosition: ItemsPosition[] = ['top', 'bottom', 'left', 'right', 'center'];
const flowDirection: FlowDirection[] = ['vertical', 'horizontal'];

export const ThemeInterface: React.SFC<CardTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.card
};

export const Basic = () => (
    <Card
        variant={select('Variant', variants, 'outlined')}
        flowDirection={select('Flow Direction', flowDirection, 'horizontal')}
        alignItems={select('Items Position', itemsPosition, 'center')}
        separator={boolean('Separator', true)}
        withPadding={boolean('WithPadding', false)}
        clickable={boolean('Clickable', false)}
        fullWidth={boolean('Full Width', false)}
        fullHeight={boolean('Full Height', false)}
    >
        <Card variant="flat">
            <Text textWeight="Strong" textSize="L3">
                8
            </Text>
            <Text>Marks</Text>
        </Card>
        <Card variant="flat" flex={6} alignItems="left">
            <Text>1760712004</Text>
            <Text textSize="L2">Stevens, Hilary</Text>
            <Text>4 Atno Avenue, Morristown, NJ, 07960</Text>
        </Card>
    </Card>
);
