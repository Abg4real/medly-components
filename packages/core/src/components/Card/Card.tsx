import { WithStyle } from '@medly-components/utils';
import React, { FC } from 'react';
import * as Styled from './Card.styled';
import { Props } from './types';

export const Card: FC<Props> & WithStyle = React.memo(React.forwardRef((props, ref) => <Styled.Card ref={ref} {...props} />));
Card.defaultProps = {
    display: 'block',
    variant: 'solid',
    flowDirection: 'vertical',
    separator: false,
    flex: 1,
    withoutPadding: false,
    alignItems: 'center',
    fullWidth: false,
    fullHeight: false
};
Card.displayName = 'Card';
Card.Style = Styled.Card;
