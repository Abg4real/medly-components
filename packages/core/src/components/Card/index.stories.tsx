import { boolean, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Text from '../Text';
import { CardStyled as Card } from './Card.styled';

storiesOf('Core', module).add('Card', () => (
    <Card width={text('Width', '250px')} height={text('Height', '250px')} center={boolean('Center', true)}>
        <Text textSize="L4" textWeight="Strong">
            55
        </Text>
        <Text textSize="L2" textWeight="Strong">
            PHARMACIES
        </Text>
    </Card>
));
