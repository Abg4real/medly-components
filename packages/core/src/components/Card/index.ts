import { CardStyled as Card } from './Card.styled';

export default Card;
