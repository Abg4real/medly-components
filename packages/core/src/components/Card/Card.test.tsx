import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Text from '../Text';
import { Card } from './Card.styled';

describe('Card component', () => {
    it('should render correctly with all the default props', () => {
        const { container } = TestUtils.render(
            <Card>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render solid variant properly ', () => {
        const { container } = TestUtils.render(
            <Card variant="solid">
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });
    it('should render flat variant properly ', () => {
        const { container } = TestUtils.render(
            <Card variant="flat">
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when it is clickable ', () => {
        const { container } = TestUtils.render(
            <Card clickable>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when fullWidth and fullHeight is set true', () => {
        const { container } = TestUtils.render(
            <Card fullWidth fullHeight>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when flow direction is horizontal and separator is set true', () => {
        const { container } = TestUtils.render(
            <Card separator flowDirection="horizontal">
                <Card separator flowDirection="horizontal" alignItems="left">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal" alignItems="right">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal" alignItems="top">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal" alignItems="bottom">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal">
                    <Text>Dummy Text</Text>
                </Card>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when separator is set true flow direction is vertical', () => {
        const { container } = TestUtils.render(
            <Card separator flowDirection="vertical" alignItems="left">
                <Card variant="flat">
                    <Text>Marks</Text>
                </Card>
                <Card variant="flat" flex={6} alignItems="right">
                    <Text>Dummy Text</Text>
                </Card>
                <Card variant="flat" flex={6} alignItems="top">
                    <Text>Dummy Text</Text>
                </Card>
                <Card variant="flat" flex={6} alignItems="bottom">
                    <Text>Dummy Text</Text>
                </Card>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });
});
