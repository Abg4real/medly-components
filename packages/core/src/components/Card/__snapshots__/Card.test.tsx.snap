// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly with all the default props 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 250px;
  height: 250px;
  background-color: #ffffff;
  border-radius: 4px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  padding: 10px;
  border: 1px solid #F9F9F9;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render correctly with all the props given 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 100px;
  height: 100px;
  background-color: #ffffff;
  border-radius: 4px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  padding: 10px;
  border: 1px solid #F9F9F9;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
    height="100px"
    width="100px"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;
